packages/docusaurus-module-type-aliases/src/index.d.ts

```diff
@@ -82,6 +82,7 @@ declare module '@generated/codeTranslations' {
 }
 
 declare module '@theme-original/*';
+declare module '@theme-init/*';
```

Comment:
FYI @Josh-Cena 

This is a bit annoying me because the `@theme-original` + `@theme-init` typedefs are incorrect atm so TS wrapped components are not typesafe

If you have TS ideas on how to support this without too much boilerplate, let me know.

We can either:
- declare 2 extra TS modules for each export
- have more complex swizzle wrap output for TS

```ts
import React, {ComponentProps} from 'react';
import type DocSidebarType from '@theme/DocSidebar';
import DocSidebar from '@theme-init/DocSidebar';

type Props = ComponentProps<typeof DocSidebarType>

export default function DocSidebarWrapper(props: Props): JSX.Element {
  return (
    <>
      <DocSidebar {...props} />
    </>
  );
}
```

I'm going to use the later for now, but ideally if we can simplify the way to obtain component props, that would be nice so open for ideas

Reply:
I think writing custom module definitions for `@theme/*` is not a sustainable idea in the first place. As we refactor our TS setup, I'd like to use path aliases that point to actual declaration files generated by `tsc`. But for now, let's just use `@theme-init/*`

Reply:
Agree, I don't like that either